File: 204_Count_Primes.cpp
#include<iostream>

using namespace std;

class Solution{
public:
    int CountPrimes(int n){
        vector<bool> primes(n, true);
        int cnt = 0;
        for (long i = 2; i < n; ++i){
            if (!primes[i]) continue;
            cnt++;
            for (long j = i; i * j < n; ++j){
                primes[i * j] = false;
            }
        }
        return cnt;
    }
};

int main(){
    Solution* solution = new Solution();

    return 0;
}

File: 204_Count_Primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """

        primes = [True] * n
        cnt = 0

        for i in xrange(2, n):
            if not primes[i]: continue
            cnt += 1
            j = i
            while i * j < n:
                primes[i * j] = False
                j += 1
        return cnt